### DISCUSSION TITLE
Different between pretrain and calling local model

### DISCUSSION THREAD

**Discussion by RickyGunawan09** [mindee/doctr#1102] [Q&A]
haii thank you for this wonderful GitHub,
sorry for up a discussion again but I get a different output between using local model and pretrain from doctr.

#pretrain
`model = ocr_predictor(det_arch='db_resnet50', reco_arch='crnn_vgg16_bn', pretrained=True,preserve_aspect_ratio=True)`

#load_from_path
```
DET_CKPT = "./models_pretrain/db_resnet50-ac60cadc.pt"
REC_CKPT = "./models_pretrain/crnn_vgg16_bn-9762b0b0.pt"

det_model = db_resnet50(pretrained=False)
rec_model = crnn_vgg16_bn(pretrained=False,vocab=VOCABS["legacy_french"])

load_det_model = torch.load(DET_CKPT, map_location="cpu")
load_rec_model = torch.load(REC_CKPT, map_location="cpu")

det_model.load_state_dict(load_det_model)
rec_model.load_state_dict(load_rec_model)

det_predictor = DetectionPredictor(PreProcessor((1024, 1024), batch_size=1, mean=(0.798, 0.785, 0.772), std=(0.264, 0.2749, 0.287)), det_model)
rec_predictor = RecognitionPredictor(PreProcessor((32, 128), preserve_aspect_ratio=True, batch_size=32, mean=(0.694, 0.695, 0.693), std=(0.299, 0.296, 0.301)), rec_model)

model = OCRPredictor(det_predictor, rec_predictor,preserve_aspect_ratio=True)
```

What's the difference?
I try to read your code but I cannot find it
thank you !! @frgfm @fg-mindee 

#using pretrain
![image](https://user-images.githubusercontent.com/65578014/200535401-25efc687-aa8b-4f44-bb4a-d84412fead6a.png)

#using local model
![image](https://user-images.githubusercontent.com/65578014/200534991-636bc3fc-7c0d-4042-9761-1aa6efd9a8fd.png)

**Comment by felixdittrich92** [comment]
Hi @RickyGunawan09 :wave:,

apologize the late response.

If i understand it correctly you used the already pretrained models but load it local !?

I tried to reproduce it but all looks fine to me:

```
import torch

from doctr.datasets import VOCABS
from doctr.io import DocumentFile
from doctr.models import ocr_predictor, crnn_vgg16_bn, db_resnet50

doc = DocumentFile.from_images("/home/felix/Desktop/1.jpg")

det_model = db_resnet50(pretrained=False)
det_model.load_state_dict(torch.load("/home/felix/.cache/doctr/models/db_resnet50-1138863a.pt", map_location="cpu"))
rec_model = crnn_vgg16_bn(pretrained=False, vocab=VOCABS["legacy_french"])
rec_model.load_state_dict(torch.load("/home/felix/.cache/doctr/models/crnn_vgg16_bn-9762b0b0.pt", map_location="cpu"))
predictor = ocr_predictor(reco_arch=rec_model, det_arch=det_model, pretrained=False)

res = predictor(doc)
res.show(doc)
```

BTW: the hash from the db_resnet50 model you used looks like an older one could you verify that it is the latest one ? 

(all downloaded models are stored in /home/user/.cache/doctr/models )
